### PR TITLE
Add mixer calibration

### PR DESCRIPTION
**Pull request by gribeill**
Mixer calibrations are now a thing in Auspex:
Run `QubitExperimentFactor.calibrate_mixer("qubit", mixer="measure")` to try it out for yourself. It will parse `measure.yml` to correctly infer which sources and AWGs are involved, and can write back to the file after the calibration.

The plotting still needs a bit of work but this contains all of the useful functionality.

### CODE REVIEW

**Comment by matthewware** [comment]
The python 3.5 CI should fail no?  If we're using formatted string literals.

**Comment by gribeill** [comment]
There was another problem too, but it should be fixed now...

File: src/auspex/mixer_calibration.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+class MixerCalibrationExperiment(Experiment):
+
+    SSB_FREQ = 10e6
+
+    amplitude = OutputConnector(unit='dBc')
+
+    I_offset = FloatParameter(default=0.0, unit="V")
+    Q_offset = FloatParameter(default=0.0, unit="V")
+    amplitude_factor = FloatParameter(default=1.0)
+    phase_skew = FloatParameter(default=0.0, unit="deg")
```
Comment (dieris) [inline src/auspex/mixer_calibration.py]: Could we make all the phase units in rad? For consistency with QGL

**Comment by gribeill** [comment]
- Yes, I convinced myself at some point this belonged in `QubitExpFactory` but I think you're right...
- I think I'll need @grahamrow 's input on why the realtime plotting isn't working
- On Orion (the computer, not the network...) I will occasionally get a single point that seems to be a "random" number, but not necessarily the first point of the sweep. Haven't found a good way to reproduce the conditions for this -- we may need to just poll the spectrum analyzer more slowly.

File: src/auspex/exp_factory.py [hunk 2]
```diff
@@ -174,9 +178,91 @@ def create(meta_file=None, expname=None, calibration=False, cw_mode=False):
...
+        """
+        spm = auspex.globals.single_plotter_mode
+        auspex.globals.single_plotter_mode = True
+
+        def sweep_offset(experiment, name, pts):
+            mce.clear_sweeps()
+            mce.add_sweep(getattr(mce, name), pts)
+            mce.run_sweeps(keep_instruments_connected = True)
+
+        offset_pts = np.linspace(-0.2, 0.2, 51)
```
Comment (dieris) [inline src/auspex/exp_factory.py]: It would be nice to have these ranges variable, i.e., part of the .yml.

**Pull request merged**